qiskit/compiler/transpiler.py

```diff
@@ -332,8 +332,15 @@ def _check_circuits_coupling_map(circuits, transpile_args, backend):
             max_qubits = parsed_coupling_map.size()
 
         # If coupling_map is None, the limit might be in the backend (like in 1Q devices)
-        elif backend is not None and not backend.configuration().simulator:
-            max_qubits = backend.configuration().n_qubits
+        elif backend is not None:
+            backend_version = getattr(backend, "version", 0)
+            if not isinstance(backend_version, int):
```

Comment:
This came up in the tests actually for the legacy backend interface all the legacy fake backends raised an exception that a bound method couldn't be compared to an int.  The cause is in `BaseBackend` `version` is an alias method for `configuration.version` which is an unused field from the iqx api payload format: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/providers/basebackend.py#L112

Reply:
I missed the name conflict when we introduced the versioned Backend interface otherwise I would have picked a different attribute for `Backend`/`BackendV1`. But this is what we'll need to do until we finish the deprecation cycle on `BaseBackend` and remove it.